src/optimizer/cardinality_estimator.cpp

```diff
@@ -518,13 +518,18 @@ idx_t CardinalityEstimator::InspectConjunctionOR(idx_t cardinality, idx_t column
 	return cardinality_after_filters;
 }
 
+
+
 idx_t CardinalityEstimator::InspectTableFilters(idx_t cardinality, LogicalOperator *op, TableFilterSet *table_filters) {
 	idx_t cardinality_after_filters = cardinality;
 	auto get = GetLogicalGet(op);
-
+	unique_ptr<BaseStatistics> column_statistics;
 	for (auto &it : table_filters->filters) {
-		auto &table_scan_bind_data = (TableScanBindData &)*get->bind_data;
-		auto column_statistics = get->function.statistics(context, &table_scan_bind_data, it.first);
+		column_statistics = nullptr;
+		if (get->bind_data && get->function.name.compare("arrow_scan") != 0) {
```

Comment:
@pedroerp @Mytherin not quite sure if this is the best way to fix the segfault. TableScanBindData doesn't exist when the table function is an arrow scan. I don't want to hardcode the string `arrow_scan` here, but it seems to be used in multiple places, so i imagine there will be a PR to fix that everywhere. 

I was wondering if you guys know of any other table functions where this might break? I couldn't find a list of the existing ones in the codebase, so wasn't sure.

Reply:
Maybe for now you can just check whether its a base table scan or not? If not, then do some default behaviour